python_modules/dagster/dagster/core/log_manager.py

```diff
@@ -180,142 +203,6 @@ def with_tags(self, **new_tags):
-
-        # Reserved keys in the message_props -- these are system generated.
-        check.invariant("orig_message" not in message_props, "orig_message reserved value")
-        check.invariant("message" not in message_props, "message reserved value")
-        check.invariant("log_message_id" not in message_props, "log_message_id reserved value")
-        check.invariant("log_timestamp" not in message_props, "log_timestamp reserved value")
-
-        # structured information about this specific message
-        dagster_message_props = DagsterMessageProps(
-            orig_message=orig_message, dagster_event=message_props.get("dagster_event")
-        )
-
-        # We first generate all props for the purpose of producing the semi-structured
-        # log message via _kv_messsage
-        all_props = dict(
-            itertools.chain(
-                self.logging_tags._asdict().items(),
-                message_props.items(),
-                dagster_message_props._asdict().items(),
-            )
-        )
-
-        # So here we use the arbitrary key DAGSTER_META_KEY to store a dictionary of
-        # all the meta information that dagster injects into log message.
-        # The python logging module, in its infinite wisdom, actually takes all the
-        # keys in extra and unconditionally smashes them into the internal dictionary
-        # of the logging.LogRecord class. We used a reserved key here to avoid naming
-        # collisions with internal variables of the LogRecord class.
-        # See __init__.py:363 (makeLogRecord) in the python 3.6 logging module source
-        # for the gory details.
-        return (
-            construct_log_string(self.logging_tags, dagster_message_props),
-            {DAGSTER_META_KEY: all_props},
-        )
-
-    def _log(self, level, orig_message, message_props):
-        """Invoke the underlying loggers for a given log level.
-
-        Args:
-            level (Union[str, int]): An integer represeting a Python logging level or one of the
-                standard Python string representations of a logging level.
-            orig_message (str): The log message generated in user code.
-            message_props (dict): Additional properties for the structured log message.
-        """
-        if not self.loggers:
-            return
-
-        level = coerce_valid_log_level(level)
-
-        message, extra = self._prepare_message(orig_message, message_props)
-
-        for logger_ in self.loggers:
-            logger_.log(level, message, extra=extra)
-
-    def log(self, level, msg, **kwargs):
-        """Invoke the underlying loggers for a given integer log level.
-
-        Args:
-            level (int): An integer represeting a Python logging level.
-            orig_message (str): The message to log.
-        """
-
-        check.str_param(msg, "msg")
-        check.int_param(level, "level")
-        return self._log(level, msg, kwargs)
-
-    def debug(self, msg, **kwargs):
```

Comment:
out of curiosity, what is the rationale for removing these individual methods and instead accepting the log level as a param to `log_dagster_event()`?

Reply:
the DagsterLogManager has two purposes -- the first is to log messages from internal sources (i.e. DagsterEvents), and the other is to allow users to add their own messages to the event log (i.e. context.log.info("my custom message") in user-provided solid code). With this diff, the DagsterLogManager inherits from the logging.Logger class, so all of its .info(), .debug(), .error() etc. functions still exist, but don't need to be specified here. In fact, you could still log a dagster event without the log_dagster_event function (it would look like log_manager.info(msg, extra={"dagster_meta": dagster_event)). However, logging dagster events this way looked a little brittle to me (if you don't know what you're doing, it's pretty easy to mess up that "extra" field, as it needs to be formatted that specific way), so for convenience I added a log_dagster_event function, with explicit parameters, to make that specific use case cleaner.